pkg/controller/job/job_controller.go

```diff
@@ -747,65 +747,67 @@ func (jm *JobController) manageJob(activePods []*v1.Pod, succeeded int32, job *b
 			utilruntime.HandleError(fmt.Errorf("More active than wanted: job %q, want %d, have %d", jobKey, wantActive, active))
 			diff = 0
 		}
```

Comment:
Is there any reason not to return here when `diff == 0`?

Reply:
Before returning, we should run the following (line 814)
```
        case err := <-errCh:
```

Reply:
The `diff == 0` case won't touch `errCh`